### ISSUE TITLE
Numeric hinting sometimes filters on numeric input

### ISSUE THREAD

**Issue by pjones** [fanglingsu/vimb#623]
### Steps to reproduce

Start `vimb` like so:

```sh
cat >test.html <<EOF
<html>
<title>Test</title>
<h1>Test</h1>
<section>
  <div role="button">Test</div>
  <div role="button">Test</div>
  <div role="button">Test</div>
  <div role="button">Test</div>
  <div role="button">Test</div>
  <div role="button">Test</div>
  <div role="button">Test</div>
  <div role="button">Test</div>
  <div role="button">Test</div>
  <div role="button">Test</div>
</section>

<script>
  let nodes = document.querySelectorAll('[role="button"]');
  for (let i=0; i<nodes.length; ++i) {
    nodes[i].addEventListener("click", function(e) {
      console.log("click", e);
    });
  }
</script>
EOF

vimb -c /dev/null -C "set hint-follow-last=false|set hint-timeout=0" test.html
```

### Expected behaviour

1. Type `f` to start hinting

2. Press any number.

3. The matching element *should* be highlighted.

### Actual behaviour

Numeric input is treated like a filter instead of selecting the hint.  For example:

1. Type `f` to start hinting

2. Press the number `9` on the keyboard

3. All hints disappear

4. The inputbox contains `;o9`

(Note, I think this might be related to #554)

### Version Information

```
Version:         3.6.0
WebKit compile:  2.26.4
WebKit run:      2.26.4
GTK compile:     3.24.13
GTK run:         3.24.13
libsoup compile: 2.68.3
libsoup run:     2.68.3
Extension dir:   /nix/store/syvjd21blzwfirpxnaf1kd7qi3yfl7ph-vimb-3.6.0/lib/vimb
```

**Comment by pjones** [comment]
I finally got around to testing this with the latest vimb commit and it seems to be fixed there.

@fanglingsu can we get a new release?

**Comment by fanglingsu** [comment]
@pjones nice to hear that it works for you now. But the only commit I can think that it fixed the issue could be related to #600. I'll try to prepare the next release soon.
